loadtest/control/simulcontroller/controller.go

```diff
@@ -302,3 +363,42 @@ func (c *SimulController) sendFailStatus(reason string) {
 func (c *SimulController) sendStopStatus() {
 	c.status <- control.UserStatus{ControllerId: c.id, User: c.user, Info: "user stopped", Code: control.USER_STATUS_STOPPED}
 }
+
+// InjectAction allows a named UserAction to be injected that is run once, at the next
+// available opportunity. These actions can be injected via the coordinator via
+// CLI or Rest API.
+func (c *SimulController) InjectAction(actionID string) control.UserActionResponse {
+	var action userAction
+	var ok bool
+
+	// include some actions that are not normally supported by SimulController
+	switch actionID {
+	case "Reload":
+		action = userAction{
+			name: "Reload",
+			run:  func(_ user.User) control.UserActionResponse { return c.reload(false) },
+		}
+	default:
+		action, ok = c.actionMap[actionID]
+		if !ok {
+			return control.UserActionResponse{
+				Info: fmt.Sprintf("Action %s not supported by SimulController", actionID),
+			}
+		}
+	}
+
+	select {
+	case c.injectedActionChan <- action:
+		return control.UserActionResponse{
+			Info: fmt.Sprintf("Action %s queued successfully", actionID),
+		}
+	case <-time.After(time.Second * 15):
```

Comment:
Fair point.  The idea around the queuing is that someone may inject a small batch of actions to perform some one-time simulation of real client activity.  Currently it will batch up to 10 actions, which seems reasonable, but going beyond that should really be done via a controller config.  So, I agree going beyond 10 actions in a short interval should fail immediately. Fixed.